src/main/java/com/wix/mysql/distribution/service/CatchAllCommandEmitter.java

```diff
@@ -31,10 +31,28 @@ public boolean matches(Version version) {
                 format("--plugin-dir=%s/lib/plugin", baseDir),
                 format("--lc-messages-dir=%s/share", baseDir),
                 format("--port=%s", config.getPort()),
+                format("--socket=%s", sockFile(exe)),
                 "--console",
                 format("--character-set-server=%s", config.getCharset().getCharset()),
                 format("--collation-server=%s", config.getCharset().getCollate()),
                 format("--default-time-zone=%s", Utils.asHHmmOffset(config.getTimeZone())));
 
     }
+
+    /**
+     * Helper for getting stable sock classPathFile. Saving to local instance variable on service start does not work due
+     * to the way flapdoodle process library works - it does all init in {@link de.flapdoodle.embed.process.runtime.AbstractProcess} and instance of
+     * {@link com.wix.mysql.MysqldProcess} is not yet present, so vars are not initialized.
+     * This algo gives stable sock classPathFile based on single executeCommands profile, but can leave trash sock classPathFiles in tmp dir.
+     * <p>
+     * Notes:
+     * .sock classPathFile needs to be in system temp dir and not in ex. target/...
+     * This is due to possible problems with existing mysql installation and apparmor profiles
+     * in linuxes.
+     */
+    private String sockFile(IExtractedFileSet exe) throws IOException {
```

Comment:
Simonas,
It's amazing how time and time again you find this 😊
On יום ו׳, 1 ביולי 2016 at 14:12 Simonas Pinevičius <
notifications@github.com> wrote:

> In
> src/main/java/com/wix/mysql/distribution/service/CatchAllCommandEmitter.java
> <https://github.com/wix/wix-embedded-mysql/pull/56#discussion_r69284574>:
>
> >                  "--console",
> >                  format("--character-set-server=%s", config.getCharset().getCharset()),
> >                  format("--collation-server=%s", config.getCharset().getCollate()),
> >                  format("--default-time-zone=%s", Utils.asHHmmOffset(config.getTimeZone())));
> >
> >      }
> > +
> > +    /**
> > +     * Helper for getting stable sock classPathFile. Saving to local instance variable on service start does not work due
> > +     * to the way flapdoodle process library works - it does all init in {@link de.flapdoodle.embed.process.runtime.AbstractProcess} and instance of
> > +     * {@link com.wix.mysql.MysqldProcess} is not yet present, so vars are not initialized.
> > +     * This algo gives stable sock classPathFile based on single executeCommands profile, but can leave trash sock classPathFiles in tmp dir.
> > +     * <p>
> > +     * Notes:
> > +     * .sock classPathFile needs to be in system temp dir and not in ex. target/...
> > +     * This is due to possible problems with existing mysql installation and apparmor profiles
> > +     * in linuxes.
> > +     */
> > +    private String sockFile(IExtractedFileSet exe) throws IOException {
>
> Use java.nio.Files
> <https://docs.oracle.com/javase/7/docs/api/java/nio/file/Files.html> for
> temp file creation
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/wix/wix-embedded-mysql/pull/56/files/9e7c60b48f45f5ef3d17b438445a5d4604e3e2ea#r69284574>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABUIF3mPPi1bXIdwIaAV19ftpNiEyVjaks5qRPYFgaJpZM4JDAIj>
> .
>
-- 

*Ittai Zeidman*

Cell: 054-6735021

40 Hanamal street, Tel Aviv, Israel


<http://www.wix.com/>